### DISCUSSION TITLE
How to get surface in INP file？

### DISCUSSION THREAD

**Discussion by smallrain1996** [neperfepx/neper#723] [Q&A] [answered]
I'm trying to get the six surfaces x0,x1,y0,y1,z0,z1 in the ABAQUS-INP file, not the set of nodes on the surface. So I blocked "-nset" and used -faset "faces", but nothing worked. Here's my test code:

```
name=test_neper_3D
number=150
mkdir $name; cd $name
neper -T -n $number -group 1 -format vtk,tess -o $name
neper -M $name.tess -format inp,msh,phase -faset "faces" -nset none -elttype hex -rcl 0.8 -o $name
```

**Comment by rquey** [comment]
If you can tell me under what format  the fasets should be written in the inp file, I can add it. An example file would be helpful.

**Reply by smallrain1996** [reply]
Thank you very much. I have provided a test case, which is obviously not a generic case, but it should be easy to see the logic.

Step1 -- Neper Code：
```
name=test_neper_3D
number=150
mkdir $name; cd $name
neper -T -n $number -group 1 \
-stattess dim,cellnb,x,y,z,size \
-statcell id,group,size,diameq,sphericity \
-format vtk,tess -o $name
neper -M $name.tess -format inp,msh,phase \
-faset "faces" -nset none \
-statmesh eltnb,nodenb,size \
-statelt id,x,y,z \
-statnode id,x,y,z \
-statelset id,group,elts \
-elttype hex -rcl 0.8 \
-elset 'Ex0:(x-length*2/3)<0',\
'Ex1:(1-length*2/3)<x',\
'Ey0:(y-length*2/3)<0',\
'Ey1:(1-length*2/3)<y',\
'Ez0:(z-length*2/3)<0',\
'Ez1:(1-length*2/3)<z' \
-o $name
```
However, a Segmentation fault occurs when the M module is executed.
Step2 -- Add the following above the "*End Part" in the INP file.
```
*Surface, type=ELEMENT, name=surf_x0
Ex0, S6
*Surface, type=ELEMENT, name=surf_x1
Ex1, S4
*Surface, type=ELEMENT, name=surf_y0
Ey0, S3
*Surface, type=ELEMENT, name=surf_y1
Ey1, S5
*Surface, type=ELEMENT, name=surf_z0
Ez0, S1
*Surface, type=ELEMENT, name=surf_z1
Ez1, S2
```
[test_neper_3D.txt](https://github.com/neperfepx/neper/files/12143912/test_neper_3D.txt)
I changed inp to txt and uploaded it.

**Reply by rquey** [reply, marked as answer]
I've fixed the seg fault (which was related to `-elset`).  However, `-faset` won't work with `-elttype hex`...

**Reply by smallrain1996** [reply]
Thank you very much
